### PR TITLE
A few improvements for the RF522 implementation

### PR DESCRIPTION
**Pull request by vkorn**
* Error handling during the init -- so we don't proceed if device is locked by something else 
* Antenna gain setter -- setting gain power 0-7
* Close method to stop SPI device -- releasing opened device

### CODE REVIEW

File: rf522/Rf522.go [hunk 1]
```diff
@@ -18,6 +19,7 @@ type RFID struct {
 	antennaGain   int
 	MaxSpeedHz    int
 	spiDev        *spi.Device
+	stop          chan bool
```
Comment (jdevelop) [inline rf522/Rf522.go]: can we use `chan interface{}` here? More idiomatic :)

**Review by jdevelop — CHANGES_REQUESTED** [review]

**Comment by jdevelop** [comment]
Also the driver was largely migrated to https://github.com/google/periph/ so it would be nice if you can send the PR there.

**Comment by vkorn** [comment]
@jdevelop changed chan to `interface{}`

I've tried periph's implementation but can't get it to work -- it simply locks the device completely upon initialization. The only way to bring it back -- reboot RPi itself. I didn't have time to look into device opening logic in periph yet.

**Comment by jdevelop** [comment]
Interesting, I ported the code for that library and it works fine there.

**Review by jdevelop — APPROVED** [review]

**Pull request merged**